### PR TITLE
Switch to ThreadPoolExecutor/ProcessPoolExecutor instead of using spawning processes

### PR DESCRIPTION
**Pull request by lgyanf**
# What does it do?

This is an attempt to fix the issue described in #221. `cron_job.py` spawns an uncontrollable number of Python processes, which leads to random OOM errors on servers with small amount of RAM.

This PR introduces a mechanism to control the amount of processes or switch to thread-based implementation to save even more memory.

### CODE REVIEW

**Comment by Fillll** [comment]
The most significant improvement in the project for the last 6 years!

File: reddit2telegram/cron_app.py [hunk 1]
```diff
@@ -31,22 +53,12 @@ def read_own_cron(own_cron_filename, config):
...
-                    supplying_process = Process(target=supply, args=(process_to_start, config))
-                    supplying_process.start()
-                    successfully_started = True
-                    break
-                else:
-                    successfully_started = False
+    random.shuffle(list_of_processes_to_start)
+    executor = _create_thread_pool(config)
+    executor.map(
+        lambda submodule_name: _process_single_entry(submodule_name, config),
```
Comment (Fillll) [inline reddit2telegram/cron_app.py:59]: Should here be 3 args?

File: reddit2telegram/cron_app.py [hunk 2]
```diff
@@ -19,6 +17,30 @@
...
+    elif pool_class == 'ProcessPoolExecutor':
+        return concurrent.futures.ProcessPoolExecutor(size)
+    else:
+        raise ValueError(f"Invalid pool class name: {pool_class}")
+
+
+def _process_single_entry(
+    submodule_name,
+    config,
+    is_test,
```
Comment (Fillll) [inline reddit2telegram/cron_app.py:35]:
```suggestion
    is_test=False,
```

File: reddit2telegram/cron_app.py (hunk 1 above)
Comment (Fillll) [reply reddit2telegram/cron_app.py:59]: Or argument by default as suggested above.

**Review by Fillll — COMMENTED** [review]
Should that be fixed before merging?

File: reddit2telegram/cron_app.py (hunk 1 above)
Comment (lgyanf) [reply reddit2telegram/cron_app.py:58]: You are right. Changed to the default argument.

**Pull request merged**